### ISSUE TITLE
Bug - [Modal] - Custom header should allow for addition of role (a11y issue)

### ISSUE THREAD

**Issue by agagancarczyk** [patternfly/patternfly#5689]
**Describe the problem**
The problem is that the document has more than one banner landmark, explicitly affecting the <header> components with the classes .pf-c-modal-box__header and .pf-c-page__header. 

The proposed solution is to allow for a custom Modal's <header> to allow for setting a `role="presentation"` for that <header>. 

**How do you reproduce the problem?**
The problem arises in the Keycloak Admin Console in places where the Modal component is used i.e Client scopes > Client scope details > Select `acr` > Scope > Assign roles to acr

Details from Deque -> https://axe.deque.com/issues/69e32a91-eb63-43c7-a841-8c2c2cc790ae

![Screenshot 2023-06-27 at 13 04 16](https://github.com/patternfly/patternfly/assets/4890675/5f01427d-6259-4444-94cc-35a1c409bd66)

**Expected behavior**
Using a custom header for Modal should allow for the addition of `role="presentation"` to address moderate a11y violation.

**Is this issue blocking you?**
Yes, we can't have Keycoak Admin Console fully accessible.

**Screenshots**
Attached above

**What is your environment?**
 - macOS: 13.2 
 - Chrome Version 114.0.5735.133 (Official Build) (x86_64)

**What is your product and what release date are you targeting?**
Keycloak Admin Console. Target date: asap

**Any other information?**

**Comment by jessiehuff** [comment]
Thanks for looking into this @agagancarczyk! Can you help me understand what you're trying to do/why you would have two banner landmarks on the page? The modal header is interpreted as a banner role because it is currently not the descendant of an article, aside, main, nav, or section. Does your UI have another banner elsewhere that wouldn't be hidden when the modal is opened? I'd love to understand why it's happening for you so that we can pursue the best possible structure for the modal. 
@thatblindgeye

**Comment by thatblindgeye** [comment]
@agagancarczyk Trying to look into this, so far what I've found is that it seems like the code internal to the PF Modal - [the code that adds and removes the `aria-hidden` attribute](https://github.com/patternfly/patternfly-react/blob/ab4a9e53ac6095708fb82928056a6dcd4ee254fe/packages/react-core/src/components/Modal/Modal.tsx#L155-L164) from elements where the Modal isn't rendered inside of - *is* running, it's just running the code from both conditionals.

Passing the PatternFly Modal in Keycloak's `EditCLientScope.tsx` (instead of RoleMapping on line ~280), the Modal renders and `aria-hidden` attributes are applied as expected.

Adding a role of presentation to the `header` element could work (the `h1` doesn't lose its semantics), but I feel like it'd be better to instead allow customizing the `.pf-c-modal-box__header` element (default to `header`, but be able to pass in `div` or `span` when necessary). Though omitting the `header` itself (or just changing it to a div/span) would be another option. @jessiehuff do you recall whether the `aria-hidden` being applied to all other elements was in part due to the modal having the `header` element, or was it simply to hide the rest of the page from AT? If it's the latter, then we probably don't even want to use `header` inside a modal (at least by default). 

I don't believe the lack of `aria-hidden` on all of the other elements in the DOM would cause issue, since the Modal is using FocusTrap which ideally would prevent navigation to those elements in the first place. At least based on testing with VO and NVDA and removing the `aria-hidden` attribute from the `#root` element in a PatternFly example. And according to [W3C dialog pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/), "The aria-modal property introduced by ARIA 1.1 replaces [aria-hidden](https://w3c.github.io/aria/#aria-hidden) for informing assistive technologies that content outside a dialog is inert."

Alternatively we could try mimicking PatternFly's code inside of Keycloak's code wherever PatternFly's Modal is being used, essentially adding a useEffect to apply the `aria-hidden="true"` attribute to elements other than the Modal container, but that'd be a bit hacky and would require knowing that the DOM would have exactly the expected structure to apply the attribute correctly.
